back/src/Model/CommunicationManager.ts

```diff
@@ -6,32 +6,40 @@ import { WebRTCState } from "./States/WebRTCState";
 import { ICommunicationManager } from "./Interfaces/ICommunicationManager";
 import { ICommunicationState } from "./Interfaces/ICommunicationState";
 import { VoidState } from "./States/VoidState";
+import { IRecordingManager, RecordingManager } from "./RecordingManager";
 
 export class CommunicationManager implements ICommunicationManager {
     private _currentState: ICommunicationState;
     private _toFinalizeState: ICommunicationState | undefined;
     private _finalizeStateTimeout: ReturnType<typeof setTimeout> | undefined;
-    private users: Map<string, SpaceUser> = new Map<string, SpaceUser>();
-    private usersToNotify: Map<string, SpaceUser> = new Map<string, SpaceUser>();
+    private _users: Map<string, SpaceUser> = new Map<string, SpaceUser>();
+    private _usersToNotify: Map<string, SpaceUser> = new Map<string, SpaceUser>();
+    private _recordingManager: IRecordingManager;
 
     constructor(private readonly space: ICommunicationSpace) {
         this._currentState = this.getInitialState();
+        this._recordingManager = new RecordingManager(this, this.space);
     }
 
     public async handleUserAdded(user: SpaceUser): Promise<void> {
-        this.users.set(user.spaceUserId, user);
+        //TODO : race condition possible ??
+        this._recordingManager.handleAddUser(user);
+        this._users.set(user.spaceUserId, user);
         const nextState = await this._currentState.handleUserAdded(user);
         if (nextState) {
             this.setState(nextState);
         }
     }
 
-    public async handleUserDeleted(user: SpaceUser): Promise<void> {
-        this.users.delete(user.spaceUserId);
+    public async handleUserDeleted(user: SpaceUser, shouldStopRecording: boolean = true): Promise<void> {
```

Comment:
I'm a bit annoyed by the lack of symmetry because of `shouldStopRecording: boolean`
I would tend to go back to one parameter method:

```
public async handleUserDeleted(user: SpaceUser): Promise<void>
```

Then, in RecordingManager, I would remove the `handleRemoveUser` and add a `getRecordingUser` method.

Finally, I would send the update of the metadata in `Space.removeUser` (not 100% sure though...)